### ISSUE TITLE
E(x)hentai broken after upgrading 1.18.4 to 1.19.0

### ISSUE THREAD

**Issue by PBXg33k** [avluis/Hentoid#1133]
**Hentoid Version**: v1.19.0 (709)

**Device/Android Version**:  Samsung Galaxy Tab A (SM-T515) Running  Android 11

**Issue details / Repro steps**:

Unable to download anything from E-Hentai nor ExHentai AFTER upgrading from 1.18.4 to 1.19.0.

There were items in the queue before upgrade, which (also) failed. Cleared the queue and added a new gallery, also failed.

Tried the following:
* Clear all caches from within app.
* Clear all cookies and re-authenticate both on ex and e.

Not sure if related, but i also tried nhentai (which used to work flawlessly),
It takes an absurd long amount of time to download a gallery, staying for a long time on "Preparing download folder"

The library consist of 10211 items.

**Stack trace / LogCat**:
```ruby
Error log : begin
Hentoid ver: 1.19.0 (709)
API: 30
Device: SM-T515
Start : 2024-01-29T22:46:20.235Z
End : 2024-01-29T22:46:20.235Z (00:00)
-----
Error log for [Ver9] Kanojo ga Kahanshin dake de Otoko o Eranjau Ohanashi Sono Ni | She Chooses Her Man Based on Her Lower Body Alone 2 [English] [SaHa] [298560@exhentai] : 1 errors
2024-01-29T23:46:20.235 Image list - [Parsing]: No images have been found. Error = https://exhentai.org/g/298560/172da79d51 @ /298560/172da79d51
Error log : end
```

@AVnetWS/admin-team

**Comment by RobbWatershed** [comment]
E-h and Exhentai do work on my test devices, even the one that had queued items. The book you provided a log for also works fine. I'm not sure what went wrong on your side.

Did you try reinstalling the app from scratch?

> Not sure if related, but i also tried nhentai (which used to work flawlessly), it takes an absurd long amount of time to download a gallery, staying for a long time on "Preparing download folder"

I don't think it's related. That very step being long means your phone might have a hard time managing your folders, given the size of your library.

**Comment by PBXg33k** [comment]
I'll try reinstalling the app, was hoping that wouldn't be necessary exactly because of the size of my library :sweat_smile: 

The tablet isn't the fastest anymore so rebuilding might take a couple of hours.
Will close this issue if that resolves the issue.

**Comment by bloodsucker3011** [comment]
Same to me in my devices. Here is the log: 

Error log : begin
Hentoid ver: 1.19.0 (709)
API: 33
Device: B2 Ultra
Start : 2024-01-30T22:49:52.989Z
End : 2024-01-30T22:49:53.985Z (00:00)
-----
Error log for è½¯èŒå…”å…”é…± - Nahida [2614884@e-hentai] : 3 errors
2024-01-30T17:49:52.989 33 - [Networking]: Network error; HTTP statusCode=0; message=Message received from https://asmrsqqftvbbqqaocggo.hath.network/om/129002836/a45cf5c3c751935aafd41eee63205b844f624d64-1115996-1920-2560-jpg/101c5c867b281fc984df0d9ca8192f9a6885c118-71099-780-1040-jpg/780/67q00zytmzuo0e198c9/0408_33.jpg : Invalid request @ https://asmrsqqftvbbqqaocggo.hath.network/om/129002836/a45cf5c3c751935aafd41eee63205b844f624d64-1115996-1920-2560-jpg/101c5c867b281fc984df0d9ca8192f9a6885c118-71099-780-1040-jpg/780/67q00zytmzuo0e198c9/0408_33.jpg
2024-01-30T17:49:53.033 33 - [Networking]: Network error; HTTP statusCode=0; message=Message received from https://asmrsqqftvbbqqaocggo.hath.network/om/129002836/a45cf5c3c751935aafd41eee63205b844f624d64-1115996-1920-2560-jpg/101c5c867b281fc984df0d9ca8192f9a6885c118-71099-780-1040-jpg/780/67q00zytmzuo0e198c9/0408_33.jpg : Invalid request @ https://asmrsqqftvbbqqaocggo.hath.network/om/129002836/a45cf5c3c751935aafd41eee63205b844f624d64-1115996-1920-2560-jpg/101c5c867b281fc984df0d9ca8192f9a6885c118-71099-780-1040-jpg/780/67q00zytmzuo0e198c9/0408_33.jpg
2024-01-30T17:49:53.985 pages - [Parsing]: The number of downloaded images (73) does not match the book's number of pages (74) @ https://e-hentai.org/g/2614884/2b51cea70c
Error log : end

**Comment by bloodsucker3011** [comment]
Your IP address has been temporarily banned for excessive pageloads. This probably means you are using automated mirroring/harvesting software or share the IP with someone who does. The ban expires in 23 hours and 8 minutes

**Comment by RobbWatershed** [comment]
You got very different logs here :

> 2024-01-30T17:49:52.989 33 - [Networking]: Network error; HTTP statusCode=0; message=Message received from https://asmrsqqftvbbqqaocggo.hath.network/om/129002836/a45cf5c3c751935aafd41eee63205b844f624d64-1115996-1920-2560-jpg/101c5c867b281fc984df0d9ca8192f9a6885c118-71099-780-1040-jpg/780/67q00zytmzuo0e198c9/0408_33.jpg : Invalid request @ https://asmrsqqftvbbqqaocggo.hath.network/om/129002836/a45cf5c3c751935aafd41eee63205b844f624d64-1115996-1920-2560-jpg/101c5c867b281fc984df0d9ca8192f9a6885c118-71099-780-1040-jpg/780/67q00zytmzuo0e198c9/0408_33.jpg
2

That's a networking issue, or a problem on the host's side

> Your IP address has been temporarily banned for excessive pageloads. This probably means you are using automated mirroring/harvesting software or share the IP with someone who does. The ban expires in 23 hours and 8 minutes

That's you being too greedy 😉 

@PBXg33k's issue is under investigation

**Comment by PBXg33k** [comment]
A small update on my end.

Instead of reinstalling the apk i cleared the cache and data, after which i reinitialized the app while having it (re)load my library.
This did not resolve the issue, i will try removing the app completly and try it again with a freshly downloaded apk when my break starts.

**Comment by PBXg33k** [comment]
That depends on your perks afaik.

I run everything on auto (2 parallel iirc) and rarely get an IP ban. 
But i have all the necessary perks for extra traffic.

**Comment by PBXg33k** [comment]
So i just performed the following steps, and unfortunately still have the same issue:

* Uninstall Hentoid
* Redownload apk from the release page
* Install Hentoid
* Create a new directory to use as library
* Authenticate with e-hentai
* Download the first best english item

**Comment by RobbWatershed** [comment]
Thanks a lot for the detailed information.

We found and fixed the problem thanks to our Discord community. A new release is on the way soon.

**Comment by RobbWatershed** [comment]
Fix is available on today's v1.19.1
